File: packages/ui-swiftui/Sources/KinetixUI/Empty.swift
//
// Empty.swift — KinetixEmpty family.
//
// Mirrors packages/ui/src/components/empty.tsx. A placeholder for a
// zero-results state — thin styled slots around a `content` view, same
// "no fixed schema" approach as KinetixCard. KinetixEmpty itself carries
// no border/background (composes cleanly inside whatever already has
// one). Gap-fill addition (not in the original Figma source) — matches
// the shadcn/ui Empty API shape.
//

import SwiftUI

public struct KinetixEmpty<Content: View>: View {
    private let content: Content

    public init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    public var body: some View {
        VStack(spacing: 24) { content } // spacing/6
            .padding(40) // p-10, not on the shared spacing scale
            .frame(maxWidth: .infinity)
            .multilineTextAlignment(.center)
    }
}

public struct KinetixEmptyHeader<Content: View>: View {
    private let content: Content

    public init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    public var body: some View {
        VStack(spacing: 8) { content } // spacing/2
    }
}

public enum KinetixEmptyMediaVariant {
    case `default`, icon
}

public struct KinetixEmptyMedia<Content: View>: View {
    @Environment(\.kinetixColors) private var colors
    private let variant: KinetixEmptyMediaVariant
    private let content: Content

    public init(variant: KinetixEmptyMediaVariant = .default, @ViewBuilder content: () -> Content) {
        self.variant = variant
        self.content = content()
    }

    public var body: some View {
        if variant == .icon {
            content
                .foregroundStyle(colors.mutedForeground)
                .padding(8) // spacing/2
                .background(colors.muted, in: Circle())
        } else {
            content
        }
    }
}

public struct KinetixEmptyTitle: View {
    private let text: String

    public init(_ text: String) {
        self.text = text
    }

    public var body: some View {
        Text(text)
            .font(.kinetixTitleSm.weight(.medium))
            .tracking(0.1)
    }
}

public struct KinetixEmptyDescription: View {
    @Environment(\.kinetixColors) private var colors
    private let text: String

    public init(_ text: String) {
        self.text = text
    }

    public var body: some View {
        Text(text)
            .font(.kinetixBodySm)
            .foregroundStyle(colors.mutedForeground)
    }
}

public struct KinetixEmptyContent<Content: View>: View {
    private let content: Content

    public init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    public var body: some View {
        VStack(spacing: 12) { content } // spacing/3
    }
}

File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Empty.kt
package com.kinetixui.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.LocalContentColor
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.CompositionLocalProvider
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.dimensionResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * KinetixEmpty family — mirrors `packages/ui/src/components/empty.tsx`.
 * A placeholder for a zero-results state — thin styled slots around a
 * `content` composable, same "no fixed schema" approach as [KinetixCard].
 * `KinetixEmpty` itself carries no border/background (composes cleanly
 * inside whatever already has one). Gap-fill addition (not in the original
 * Figma source) — matches the shadcn/ui Empty API shape.
 */
@Composable
fun KinetixEmpty(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier.padding(40.dp), // p-10, not on the shared spacing scale
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.spacing_6)),
    ) {
        content()
    }
}

@Composable
fun KinetixEmptyHeader(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.spacing_2)),
    ) {
        content()
    }
}

enum class KinetixEmptyMediaVariant { Default, Icon }

/** `Icon` gives a muted circular badge behind the icon; `Default` renders `content` as-is. */
@Composable
fun KinetixEmptyMedia(
    variant: KinetixEmptyMediaVariant = KinetixEmptyMediaVariant.Default,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    val colors = KinetixColorScheme.current
    if (variant == KinetixEmptyMediaVariant.Icon) {
        Box(
            modifier = modifier
                .background(colors.muted, CircleShape)
                .padding(dimensionResource(R.dimen.spacing_2)),
            contentAlignment = Alignment.Center,
        ) {
            CompositionLocalProvider(LocalContentColor provides colors.mutedForeground) {
                content()
            }
        }
    } else {
        Box(modifier = modifier, contentAlignment = Alignment.Center) { content() }
    }
}

@Composable
fun KinetixEmptyTitle(text: String, modifier: Modifier = Modifier) {
    Text(
        text = text,
        modifier = modifier,
        fontWeight = FontWeight.Medium,
        fontSize = dimensionResource(R.dimen.font_size_title_sm).value.sp,
        lineHeight = dimensionResource(R.dimen.line_height_title_sm).value.sp,
        letterSpacing = dimensionResource(R.dimen.letter_spacing_wide_1).value.sp,
        textAlign = TextAlign.Center,
    )
}

@Composable
fun KinetixEmptyDescription(text: String, modifier: Modifier = Modifier) {
    val colors = KinetixColorScheme.current
    Text(
        text = text,
        modifier = modifier,
        color = colors.mutedForeground,
        fontSize = dimensionResource(R.dimen.font_size_body_sm).value.sp,
        lineHeight = dimensionResource(R.dimen.line_height_body_sm).value.sp,
        textAlign = TextAlign.Center,
    )
}

@Composable
fun KinetixEmptyContent(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(dimensionResource(R.dimen.spacing_3)),
    ) {
        content()
    }
}
